### DISCUSSION TITLE
Running pipeline on already processed subjects after manually correcting MELODIC outputs

### DISCUSSION THREAD

**Discussion by dsodums** [McIntosh-Lab/tvb-ukbb#166] [Help]
Hi there,

I would like to test out the impact of manually correcting some MELODIC outputs for some cases that had the pipeline run on it. If I manually change the fix4melview_Training_Rockland_thr30.txt file, what would be the best way to temporarily edit the pipeline code so it starts at that step in the pipeline?

Thanks,
Devin

**Comment by justinliuyuwang** [comment]
Hi Devin,

Once you've manually changed the fix4melview_Training_Rockland_thr30.txt file, you will need to edit 

```${FSL_FIXDIR}/fix . ${FSL_FIXDIR}/training_files/${TRAINING_FILE} 20 -m -h 100``` 

in [Line 28 of bb_functional_pipeline/bb_fix](https://github.com/McIntosh-Lab/tvb-ukbb/blob/e917c1100e83cab239809f902116f9c19434366e/bb_functional_pipeline/bb_fix#L28) to be 

```${FSL_FIXDIR}/fix -a ${MELOUT}/fix4melview_${TRAIN}_thr${THRESH}.txt . ${FSL_FIXDIR}/training_files/${TRAINING_FILE} 20 -m -h 100```


Note: These instructions assume that your TRAIN and THRESH env variables have not changed (i.e. you are only changing the fix4melview_Training_Rockland_thr30.txt file contents and not filename).


Furthermore, you may need to comment out lines 76 to 97 and lines 105 to 131 of [bb_functional_pipeline/bb_pipeline_func.py](https://github.com/McIntosh-Lab/tvb-ukbb/blob/main/bb_functional_pipeline/bb_pipeline_func.py) to suppress previous steps of the rfMRI pipeline as well as lines 113 to 122, and 140 of [bb_pipeline_tools/bb_pipeline.py](https://github.com/McIntosh-Lab/tvb-ukbb/blob/Cam-CAN/bb_pipeline_tools/bb_pipeline.py) to suppress structural and diffusion processing. This may not be necessary, but it should speed up processing.

Hope that helps!

**Reply by justinliuyuwang** [reply]
@dsodums did this end up working out for you?

**Reply by justinliuyuwang** [reply]
Sorry... there is a possibility the line of code I provided may not work. If not, then please try this:

```
${FSL_FIXDIR}/fix -a <path>/<to>/fix4melview_Training_Rockland_thr30.txt -m -h 100
```

**Reply by dsodums** [reply]
Before I try this out, am I correct to understand that the "<path>/<to>/" part needs to be changed to the specific path for a specific subject? If so, I would need to change it each time I test this out on a different subject?

**Reply by dsodums** [reply]
Sorry the part I was referring to didn't get included:

"<path>/<to>"

**Reply by justinliuyuwang** [reply]
Thanks to you making this very relevant discussion post, we created an up-to-date wiki page for your current problem: [Manual Classification of ICs](https://github.com/McIntosh-Lab/tvb-ukbb/wiki/Modifying-FEAT,-MELODIC-ICA,-FIX#manual-classification-of-ics) 

For generalizability and code-automation it is recommended that you have your manual labelled IC file placed in the corresponding ICA directory and `<path>/<to>/<your_manual_labelled_IC_file>.txt` can just be the filename of the manual labelled IC file (e.g. `Rockland_classified_ICs.txt`). This way you don't need to change your code with each subject because the IC file name is generic.

**Reply by dsodums** [reply]
This is amazing, thanks so much! Happy to help, even if it's just raising issues lol. I'll try this out and let you know how it goes.

**Comment by dsodums** [comment]
@justinliuyuwang sorry about that, I was on vacation and haven't had a chance to test it out. I'll give it a go this week and will let you know, thanks for sending the updated code!

**Reply by justinliuyuwang** [reply]
No worries, didn't mean to rush you! Hope you had a great vacation :)
